GitHubTeams.ps1

```diff
@@ -276,6 +315,488 @@ filter Get-GitHubTeamMember
     return (Invoke-GHRestMethodMultipleResult @params | Add-GitHubUserAdditionalProperties)
 }
 
+function New-GitHubTeam
+{
+<#
+    .SYNOPSIS
+        Creates a team within an organization on GitHub.
+
+    .DESCRIPTION
+        Creates a team within an organization on GitHub.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OrganizationName
+        The name of the organization to create the team in.
+
+    .PARAMETER TeamName
+        The name of the team.
+
+    .PARAMETER Description
+        The description for the team.
+
+    .PARAMETER MaintainerName
+        A list of GitHub IDs for organization members who will become team maintainers.
+
+    .PARAMETER RepositoryFullName
+        The full name (e.g., "organization-name/repository-name") of repositories to add the team to.
+
+    .PARAMETER Privacy
+        The level of privacy this team should have.
+
+    .PARAMETER ParentTeamName
+        The name of a team to set as the parent team.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .INPUTS
+        System.String
+        GitHub.User
+
+    .OUTPUTS
+        GitHub.Team
```

Comment:
```suggestion
        GitHub.TeamDetail
```

Reply:
I've decided to swap this round and return a type of `GitHub.TeamSummary` when `Get-GitHubTeam` is called without specifiying a `TeamName` and return a type of `GitHub.Team` for when a `TeamName` is specified, and for the output of the `New` and `Set` functions.